### PR TITLE
Issue with link processing by hook/listener in BE 

### PR DESCRIPTION
**Pull request by tastendruecker**
Hi Sebastian,

we found a small issue with EXT:content_slug in the TYPO3 BE module "redirects". If there are targets pointing to links like "t3://page?uid=123#456", the hook/listener classes will be invoked and produce an exception in the core's ContentObjectRenderer class. The latter relies on TSFE which is obviously not available.

So, I'd like to suggest checking the availablity of TSFE before processing links...

Regards,
Sven

### CODE REVIEW

**Comment by sebkln** [comment]
Hi Sven, thanks for your report and PR! I will have a look at it soon.

**Comment by sebkln** [comment]
I checked the behavior of redirects with fragments in the target, both with and without your PR.

Checking for `$GLOBALS['TSFE']` didn't fix the issues in my test installations. Instead, `$GLOBALS['TSFE']` will trigger a PHP warning with PHP 8.x as soon as I load the Redirects module in TYPO3 v11 or v12:

    PHP Warning: Undefined global variable $TSFE in …/Hooks/ReplaceFragment.php (or …/Listener/ModifyFragment.php)

Long story short: All exceptions are related to symfony conditions in the first found rootpage in the page tree; and are triggered if a request is missing.

I now created a branch where I check for a given request: https://github.com/sebkln/content_slug/tree/bugfix/redirects
My first tests are promising. Still need to check it properly, though.

Can you check if this change solves the issue in your installation?

**Comment by sebkln** [comment]
Fixed with 12cf46c1750866e17fc0cc631fb1baebc767665c.
